### PR TITLE
Time series profile

### PR DESCRIPTION
**Pull request by tmgordeeva**
This doesn't add complete profiling to time series, but enables profiling in a few more locations.

### CODE REVIEW

**Review by martijnvg — COMMENTED** [review]
In general this looks good, maybe also add a yaml rest that verifies that profiling a search request with time_series aggregation doesn't fail?

Also looks like spotlessApply should be run to address code style issues.

File: server/src/main/java/org/elasticsearch/search/internal/ContextIndexSearcher.java [hunk 1]
```diff
@@ -63,6 +63,15 @@ public class ContextIndexSearcher extends IndexSearcher implements Releasable {
     private QueryProfiler profiler;
     private final MutableQueryTimeout cancellable;
 
+    // Constructor which doesn't wrap and doesn't throw an exception.
+    public ContextIndexSearcher(IndexReader reader, Similarity similarity, QueryCache queryCache, QueryCachingPolicy queryCachingPolicy) {
```
Comment (martijnvg) [inline server/src/main/java/org/elasticsearch/search/internal/ContextIndexSearcher.java:67]: Why is this new constructor needed? Can the existing public  constructor be used here (with `wrapWithExitableDirectoryReader` set to `false`)?

**Review by martijnvg — COMMENTED** [review]
Looks good, left one question.

**Comment by tmgordeeva** [comment]
@martijnvg does this change look alright to merge? It should be similar to what we discussed, with tests.

**Comment by martijnvg** [comment]
> does this change look alright to merge? It should be similar to what we discussed, with tests.

This change does look good. I just I have one question about the new constructor added with this PR.
I think we can reuse the existing constructor? We would need to catch IOException in `TimeSeriesIndexSearcher` and set `wrapWithExitableDirectoryReader` to false in the current constructor of `ContextIndexSearcher`.

**Review by martijnvg — APPROVED** [review]
LGTM

**Pull request merged**